src/components/TodoModal/TodoModal.tsx

```
@@ -1,42 +1,87 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  const handleResetCurrentTodo = () => {
+    dispatch(currentTodoSlice.actions.reset());
+  };
+
+  useEffect(() => {
+    if (!currentTodo) {
+      return;
+    }
+
+    setLoading(true);
+    setUser(null);
+
+    getUser(currentTodo.userId)
+      .then(setUser)
+      .catch(() => setUser(null))
```

Improve error handling: the `.catch` currently swallows the error and only sets user to null. Log the error to help debugging, e.g. `catch(err => { console.error(err); setUser(null); })`. This provides valuable context when requests fail without changing user-facing behavior.